### PR TITLE
yarn: remove `bottle :unneeded`

### PR DESCRIPTION
**Pull request by Amar1729**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

See #75943

### CODE REVIEW

**Comment by Amar1729** [comment]
```
> sha256sum *.tar.gz

3de51a269f5e18428745059a001372566190386deb78c928960b16802006bb20  yarn--1.22.10.arm64_big_sur.bottle.tar.gz
6088dffecef6ff3fe5f307f01a2a0442ee20247e759757b340dce06ec1a5bee1  yarn--1.22.10.big_sur.bottle.tar.gz
6088dffecef6ff3fe5f307f01a2a0442ee20247e759757b340dce06ec1a5bee1  yarn--1.22.10.catalina.bottle.tar.gz
04092f1f0ddf6d3981c9e25b2ac1a9baaaeb182507e1be0d4e94e89b40892c67  yarn--1.22.10.mojave.bottle.tar.gz
```

```
> diff -r yarn--1.22.10.arm64_big_sur.bottle yarn--1.22.10.big_sur.bottle

diff -r yarn--1.22.10.arm64_big_sur.bottle/yarn/1.22.10/libexec/lib/cli.js yarn--1.22.10.big_sur.bottle/yarn/1.22.10/libexec/lib/cli.js
2216c2216
< var POSIX_GLOBAL_PREFIX = exports.POSIX_GLOBAL_PREFIX = `${process.env.DESTDIR || ''}/usr/local`;
---
> var POSIX_GLOBAL_PREFIX = exports.POSIX_GLOBAL_PREFIX = `${process.env.DESTDIR || ''}@@HOMEBREW_PREFIX@@`;
31583c31583
<     // /usr/local/bin/node --> prefix=/usr/local
---
>     // @@HOMEBREW_PREFIX@@/bin/node --> prefix=@@HOMEBREW_PREFIX@@
31825c31825
<         // E.g. /usr/local/share vs /root on linux machines, check the additional location
---
>         // E.g. @@HOMEBREW_PREFIX@@/share vs /root on linux machines, check the additional location
34675c34675
< var userHomeDir = (_rootUser || _load_rootUser()).default ? path.resolve('/usr/local/share') : home;
---
> var userHomeDir = (_rootUser || _load_rootUser()).default ? path.resolve('@@HOMEBREW_PREFIX@@/share') : home;
```

Another case of homebrew prefix getting changed out with `@@HOMEBREW_PREFIX@@` on x86, not on ARM.

unpacked x86 big sur and mojave bottles don't differ.

**Comment by Amar1729** [comment]
After rebuild:
```
sha256sum *.tar.gz

ae0054d3e16c5e2b209efad5bce26d9d7c180f4fa21df221085988e32dcc686d  yarn--1.22.10.arm64_big_sur.bottle.tar.gz
55ea410eac91e3f216f5a374bbd23f0d57463a52b79b26eed29b3de2b56503e8  yarn--1.22.10.big_sur.bottle.tar.gz
55ea410eac91e3f216f5a374bbd23f0d57463a52b79b26eed29b3de2b56503e8  yarn--1.22.10.catalina.bottle.tar.gz
55ea410eac91e3f216f5a374bbd23f0d57463a52b79b26eed29b3de2b56503e8  yarn--1.22.10.mojave.bottle.tar.gz
```

**Comment by carlocab** [comment]
This is related to `brew` relocation code. Similar to #75967. If there aren't too many files with `/usr/local` references, then this can be easily fixed with an
```ruby
inreplace file, "/usr/local", HOMEBREW_PREFIX
```

File: Formula/yarn.rb [hunk 1]
```diff
@@ -24,6 +22,7 @@ def install
     (bin/"yarnpkg").write_env_script "#{libexec}/bin/yarn.js",
       PREFIX:            HOMEBREW_PREFIX,
       NPM_CONFIG_PYTHON: "/usr/bin/python"
+    inreplace "#{libexec}/lib/cli.js", "/usr/lib", HOMEBREW_PREFIX
```
Comment (carlocab) [inline Formula/yarn.rb:25]: Should this be
```suggestion
    inreplace "#{libexec}/lib/cli.js", "/usr/local", HOMEBREW_PREFIX
```
? `brew` doesn't replace `/usr/lib`. Squash your commits together with the first when you make this change please. Thanks!

Comment (Amar1729) [reply Formula/yarn.rb:25]: oof thanks for the catch, i was just on autopilot when i typed that

**Comment by Amar1729** [comment]
Hashes match now:
```
sha256sum *.tar.gz

5c8f50922a573a4fc2f15b2b5b0fb457d93312474f85b1a23dd3af7d3e072650  yarn--1.22.10.arm64_big_sur.bottle.tar.gz
5c8f50922a573a4fc2f15b2b5b0fb457d93312474f85b1a23dd3af7d3e072650  yarn--1.22.10.big_sur.bottle.tar.gz
5c8f50922a573a4fc2f15b2b5b0fb457d93312474f85b1a23dd3af7d3e072650  yarn--1.22.10.catalina.bottle.tar.gz
5c8f50922a573a4fc2f15b2b5b0fb457d93312474f85b1a23dd3af7d3e072650  yarn--1.22.10.mojave.bottle.tar.gz
```

**Review by carlocab — APPROVED** [review]
Thanks @Amar1729!

**Pull request closed without merging**